tests/kani/AsyncAwait/manual_spawn.rs

```diff
@@ -160,32 +247,34 @@ pub fn spawnable_block_on<F: Future<Output = ()> + Sync + 'static>(
     }
 }
 
-struct YieldNow {
-    yielded: bool,
-}
+/// Suspends execution of the current future, to allow the scheduler to poll another future
+///
+/// Specifically, it returns a future that
+pub fn yield_now() -> impl Future<Output = ()> {
+    struct YieldNow {
+        yielded: bool,
+    }
 
-impl Future for YieldNow {
-    type Output = ();
+    impl Future for YieldNow {
+        type Output = ();
 
-    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> std::task::Poll<Self::Output> {
-        if self.yielded {
-            cx.waker().wake_by_ref(); // For completeness. But Kani currently ignores wakers.
-            std::task::Poll::Ready(())
-        } else {
-            self.yielded = true;
-            std::task::Poll::Pending
+        fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> std::task::Poll<Self::Output> {
+            if self.yielded {
+                cx.waker().wake_by_ref(); // For completeness. But Kani currently ignores wakers.
+                std::task::Poll::Ready(())
+            } else {
+                self.yielded = true;
+                std::task::Poll::Pending
+            }
         }
     }
-}
 
-/// Suspends execution of the current future, to allow the scheduler to poll another future
-pub fn yield_now() -> impl Future<Output = ()> {
     YieldNow { yielded: false }
 }
 
 #[kani::proof]
 #[kani::unwind(4)]
-fn arc_spawn_deterministic_test() {
+fn deterministic_schedule() {
     let x = Arc::new(AtomicI64::new(0)); // Surprisingly, Arc verified faster than Rc
     let x2 = x.clone();
     spawnable_block_on(
```

Comment:
isn't this duplicating the harness from `spawn.rs`?

Reply:
see https://github.com/model-checking/kani/pull/1659#discussion_r1104463102

Reply:
removed as discussed above